packages/experimental/AppearanceAdditions/ios/FRNAppearanceAdditions.m

```diff
@@ -137,9 +134,20 @@ - (void)stopObserving {
 
 - (void)appearanceChanged:(NSNotification *)notification {
     if (_hasListeners) {
+        RCTBridge *notificationBridge = [[notification object] bridge];
+        if (![notificationBridge isKindOfClass:[RCTBridge class]]) {
+            return;
+        }
+
+        // Don't send the appearanceChanged event if the notification didn't originate from the same react native instance
+        RCTBridge *currentBridge = [[self bridge] parentBridge];
+        if (![currentBridge isEqual:notificationBridge]) {
+            return;
+        }
+
         UITraitCollection *traitCollection = [[notification userInfo] valueForKey:RCTUserInterfaceStyleDidChangeNotificationTraitCollectionKey];
         if (![traitCollection isKindOfClass:[UITraitCollection class]]) {
-            traitCollection = nil;
+            return;
```

Comment:
Should we do some sort of assert or log here?

Reply:
I don't expect us to ever get into this state, It shouldn't be possible. I can add an assert if you think there should still be on. There's similar logic in the RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD calls as well (just flipped), do you think it should be added there as well?

Reply:
That's a very heavily debated issue - personally I'm a bigger fan of failing safe, like here, though if there was some way to log that'd be nice. But personally I wouldn't crash.

Reply:
I'll start a conversation about logging in RN because I haven't seen any usages of it, and can revisit this